src/charm.py

```diff
@@ -950,12 +959,17 @@ def _initialise_replica_set(self, event: StartEvent) -> None:
                 self._peers.data[self.app]["replica_set_hosts"] = json.dumps(
                     [self._unit_ip(self.unit)]
                 )
+
                 logger.info("User initialization")
                 self._init_operator_user()
                 self._init_backup_user()
                 self._init_monitor_user()
-                logger.info("Manage relations")
-                self.client_relations.oversee_users(None, None)
+
+                # in sharding, user management is handled by mongos subordinate charm
+                if self.is_role(Config.REPLICATION):
```

Comment:
nits: maybe we can use a constant named "REPLICA" or similar (because "REPLICATION" looks like action) ?

Reply:
I was keeping the name REPLICATION because it is consistent with the spec doc's naming of roles. (i.e. the role name is `replication`, so I think the config option should use the should use the same name